_pytest/assertion/__init__.py

```diff
@@ -166,10 +177,18 @@ def warn_about_missing_assertion(mode):
         else:
             specifically = "failing tests may report as passing"
 
-        sys.stderr.write("WARNING: " + specifically +
-                         " because assert statements are not executed "
-                         "by the underlying Python interpreter "
-                         "(are you using python -O?)\n")
+        # temporarily disable capture so we can print our warning
+        capman = pluginmanager.getplugin('capturemanager')
+        try:
+            out, err = capman.suspendcapture()
+            sys.stderr.write("WARNING: " + specifically +
+                             " because assert statements are not executed "
+                             "by the underlying Python interpreter "
+                             "(are you using python -O?)\n")
+        finally:
+            capman.resumecapture()
+            sys.stdout.write(out)
+            sys.stderr.write(err)
```

Comment:
What's the point of writing the captured data out again after resuming capturing? Wouldn't it be captured and double then?